src/VisualStudio/Core/Def/Implementation/Experimentation/AnalyzerVsixSuggestedActionCallback.cs

```diff
@@ -0,0 +1,234 @@
+
+            // If the user has previously clicked don't show again, then we bail out immediately
+            if (options.GetOption(AnalyzerABTestOptions.NeverShowAgain))
+            {
+                // We set _infoBarShown to true here to make sure that we never run these checks again
+                lock(_scheduleLock)
+                {
+                    _checksRunning = false;
+                    _infoBarShown = true;
+                }
+
+                return;
+            }
+
+            // Filter for valid A/B test candidates. Candidates fill the following critera:
+            //     1: Are a Dotnet user (as evidenced by the fact that this code is being run)
+            //     2: Have triggered a lightbulb on 3 separate days
+
+            // If the user hasn't met candidacy conditions, then we check them. Otherwise, proceed to info bar check
+            var isCandidate = options.GetOption(AnalyzerABTestOptions.HasMetCandidacyRequirements);
+            if (!isCandidate)
+            {
+                // We store in UTC to avoid any timezone offset weirdness
+                var lastTriggeredTime = DateTime.FromBinary(options.GetOption(AnalyzerABTestOptions.LastDayUsedSuggestionAction));
+                var currentTime = DateTime.UtcNow;
+                var span = currentTime - lastTriggeredTime;
+                if (span.TotalDays >= 1)
+                {
+                    options = options.WithChangedOption(AnalyzerABTestOptions.LastDayUsedSuggestionAction, currentTime.ToBinary());
+                    var usageCount = options.GetOption(AnalyzerABTestOptions.UsedSuggestedActionCount);
+                    usageCount++;
+                    options = options.WithChangedOption(AnalyzerABTestOptions.UsedSuggestedActionCount, usageCount);
+
+                    if (usageCount >= 3)
+                    {
+                        isCandidate = true;
+                        options = options.WithChangedOption(AnalyzerABTestOptions.HasMetCandidacyRequirements, true);
+                    }
+                    _workspace.Options = options;
+                }
+            }
+
+            // If the user still isn't a candidate, then return. Otherwise, we move to checking if the infobar should be shown
+            if (!isCandidate)
+            {
+                lock (_scheduleLock)
+                {
+                    _checksRunning = false;
+                }
+
+                return;
+            }
+
+            ShowInfoBarIfNecessary();
+        }
+
+        private void ShowInfoBarIfNecessary()
+        {
+            if (_experimentationService.IsExperimentEnabled(AnalyzerEnabledFlight))
+            {
+                // If we got true from the experimentation service, then we're in the treatment group, and the experiment is enabled.
+                // We determine if the infobar has been displayed in the past 24 hours. If it hasn't been displayed, then we do so now.
```

Comment:
Only if they dismiss the bar without installing or saying never show again, and they'll only see it once every 24 hours at most.